drift-codec/src/main/java/io/airlift/drift/codec/internal/compiler/ThriftCodecByteCodeGenerator.java

```diff
@@ -1038,8 +1044,10 @@ private static boolean needsCodec(ThriftFieldMetadata fieldMetadata)
 
     private static ParameterizedType toCodecType(ThriftStructMetadata metadata)
     {
-        String className = type(metadata.getStructClass()).getClassName();
-        return typeFromPathName(PACKAGE + "/" + className + "Codec");
+        String className = type(metadata.getStructClass()).getClassName()
+                + "_" + Instant.now().atZone(UTC).format(TIMESTAMP_FORMAT)
```

Comment:
We added timestamp to Presto since generates ephemeral classes for each query (or parts of a query), but we shouldn't need it in Drift as these classes are permanent (a limited set and cached forever).

Reply:
We have used it so as to generate a unique class name as `type(metadata.getStructClass()).getClassName()` won't capture the generic parameter. Now have replaced with `metadata.getStructType().getTypeName().replaceAll("[^a-zA-Z0-9]+", "_");
`